src/DssSpell.t.sol

```diff
@@ -1755,20 +1751,22 @@ contract DssSpellTest is DSTest, DSMath {
                 assertTrue(dunk >= RAD && dunk < MILLION * RAD, string(abi.encodePacked("TestError/cat-dunk-range-", ilk)));
 
                 (address flipper,,) = cat.ilks(ilk);
-                FlipAbstract flip = FlipAbstract(flipper);
-                // Convert BP to system expected value
-                uint256 normalizedTestBeg = (values.collaterals[ilk].flip_beg + 10000)  * 10**14;
-                assertEq(uint256(flip.beg()), normalizedTestBeg, string(abi.encodePacked("TestError/flip-beg-", ilk)));
-                assertTrue(flip.beg() >= WAD && flip.beg() <= 110 * WAD / 100, string(abi.encodePacked("TestError/flip-beg-range-", ilk))); // gte 0% and lte 10%
-                assertEq(uint256(flip.ttl()), values.collaterals[ilk].flip_ttl, string(abi.encodePacked("TestError/flip-ttl-", ilk)));
-                assertTrue(flip.ttl() >= 600 && flip.ttl() < 10 hours, string(abi.encodePacked("TestError/flip-ttl-range-", ilk)));         // gt eq 10 minutes and lt 10 hours
-                assertEq(uint256(flip.tau()), values.collaterals[ilk].flip_tau, string(abi.encodePacked("TestError/flip-tau-", ilk)));
-                assertTrue(flip.tau() >= 600 && flip.tau() <= 3 days, string(abi.encodePacked("TestError/flip-tau-range-", ilk)));          // gt eq 10 minutes and lt eq 3 days
-
-                assertEq(flip.wards(address(flipMom)), values.collaterals[ilk].flipper_mom, string(abi.encodePacked("TestError/flip-flipperMom-auth-", ilk)));
-
-                assertEq(flip.wards(address(cat)), values.collaterals[ilk].liqOn ? 1 : 0, string(abi.encodePacked("TestError/flip-liqOn-", ilk)));
-                assertEq(flip.wards(address(pauseProxy)), 1, string(abi.encodePacked("TestError/flip-pause-proxy-auth-", ilk))); // Check pause_proxy ward
+                if (flipper != address(0)) {
```

Comment:
We should see when this conditional was introduced as in theory we shouldn't need at all. If we enter via `if (values.collaterals[ilk].liqType == "flip") {` we should have a flipper, otherwise the collateral is uncorrectly set. I think in this case the actual mainnet is actually the correct case.

Reply:
I agree as the flipper address is returned from the `cat`:
`(address flipper,,) = cat.ilks(ilk);`
so if not present, the collateral values are then set incorrectly as Gonza mentioned.
We could keep it eventually as an extra sanity check but then adding it also for the clipper address returned from the `dog` as well in the tests.

Reply:
Well I think it's the opposite. If you want a sanity check then things should fail if the flipper isn't present which is not the case putting the conditional. Excepting you meant that.

Reply:
Oh! right, sorry I meant we could replace it in favour of a sanity check for the flipper and clipper addresses like:
`assertTrue(flipper != 0);`
`assertTrue(clipper != 0);
respectively

Reply:
Yeah that would be fine, but put an error message please.

Reply:
```
assertTrue(flipper != address(0), string(abi.encodePacked("TestError/invalid-flipper-address-", flipper));
assertTrue(clipper != address(0), string(abi.encodePacked("TestError/invalid-clipper-address-", clipper));
```

Reply:
In the error message, where it says `flipper/clipper` should be the `ilk`.

Reply:
Oh, right! Updated.

Reply:
fixed, thank you